src/ui/server/shared-validation/alpha-numerical-characters-only/index.ts

```diff
@@ -17,33 +18,29 @@ import generateValidationErrors from '../../helpers/validation';
 const alphaNumericalCharactersOnlyValidation = (fieldValue: string, fieldId: string, errorMessage: string, errors: object) => {
   /**
    * Check the string contains:
-   * 1) Uppercase alpha characters
-   * 2) numerical characters
+   * 1) alpha characters
+   * 2) alpha characters and numerical characters
+   * 3) does not contain only numerical characters
+   * 4) does not contain only special characters
    */
-  const alphaCharacters = fieldValue.match(REGEX.INCLUDES_UPPERCASE_ALPHA_CHARACTERS);
-  const numericalCharacters = fieldValue.match(REGEX.INCLUDES_NUMERICAL_CHARACTERS);
+  const joiString = Joi.string();
 
-  /**
-   * If the string contains alpha and numerical characters, generate a total length of these valid characters.
-   * NOTE: Such characters can be positioned differently, for example, alpha characters in "AB7CD" would return an array with 2x items.
-   * We therefore need to use .join('') to create a single item for the appropriate characters.
-   * If this length does NOT match the total fieldValue length,
-   * an invalid character has been provided, e.g an empty space or special character.
-   * Anything other than an alpha or numerical character is invalid.
-   */
-  if (alphaCharacters && numericalCharacters) {
-    const validCharacters = [...alphaCharacters.join(''), ...numericalCharacters.join('')];
+  const alphaNumSchema = () => joiString.alphanum().required();
+  const numbersOnlySchema = () => Joi.string().pattern(REGEX.NUMERICAL_CHARACTERS);
 
-    if (validCharacters.length === fieldValue.length) {
-      return false;
-    }
-  }
+  const validationAlphaNum = alphaNumSchema().validate(fieldValue);
+  const validationNumbersOnly = numbersOnlySchema().validate(fieldValue);
 
   /**
-   * Otherwise, the provided fieldValue is invalid.
-   * Return validation errors.
+   * if the field value does not match the alpha numerical characters validation
+   * or if the field value does not contain an error for only numerical values (string is numbers only)
+   * then return generate and return validation errors
    */
-  return generateValidationErrors(fieldId, errorMessage, errors);
+  if (validationAlphaNum.error || !validationNumbersOnly.error) {
```

Comment:
Why does `validationNumbersOnly` have a `!`?

Reply:
in the documentation above.  If validationNumbersOnly does not return a number - means that value is a number so invalid

Reply:
Finding this hard to read/understand even with the comments and re-reading, any way we can make it clearer?

Reply:
I think it's this part that could be clearer:

```ts
!validationNumbersOnly.error
```

It reads "if there is no error" ... return an error